sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -232,12 +230,17 @@ private void performLogging(LogLevel logLevel, LogLevel environmentLogLevel, boo
      * Determines if the environment and logger support logging at the given log level.
      *
      * @param logLevel Logging level for the log message.
-     * @param environmentLoggingLevel Logging level the environment is set to support.
      * @return Flag indicating if the environment and logger are configured to support logging at the given log level.
      */
-    private boolean canLogAtLevel(LogLevel logLevel, LogLevel environmentLoggingLevel) {
+    public boolean canLogAtLevel(LogLevel logLevel) {
+        loadSystemLogLevel();
```

Comment:
Is this `load` operation necessary each time this method is called?

Reply:
It is a lazy initialization. It will check rootLogLevel exist or not in loadSystemLogLevel.

Reply:
Make the call every time instead of storing them.